### DISCUSSION TITLE
Strommessgenauigkeit beim JK BMS

### DISCUSSION THREAD

**Discussion by SW-Niko** [hoylabs/OpenDTU-OnBattery#1361] [Q&A] [answered]
Hallo,
ich möchte an meiner Anlage den Batteriestrom messen und sehe 2 Optionen für mich.
1. Ich baue einfach einen Victron Smart Shunt ein.
2. Ich ersetze mein "passives nicht auslesbares" BMS durch ein "aktives und auslesbares" BMS.

Aktuell tendiere ich eher zu Option 2. In den technischen Daten von JK kann ich aber nichts zur Strommessgenauigkeit finden.
Hat hier im Forum jemand Erfahrung damit? Mit einer Messtoleranz von 2-5% kann ich leben.
Und wie weit kann ich runter messen? 100mA, 10mA oder noch kleiner? 

Vielen Dank für eure Unterstützung.

**Comment by DonJohnLong** [comment]
Hallo Niko

Wenn die Batteriespannung und der Stromfluss kalibriert wurden, sind meine zwei jk bms zimmlich genau.

Ich habe meine mit einem labornetzteil auf 3A kalibriert.

Wie genau dies dann aber ist kann ich nicht 100% sagen, passt aber zu den daten der victrons und des hoymiles.

Hier noch ein video dass ich letztens zum topic gesehen habe: https://youtu.be/FUHOSjIJY0Y

Lg

**Reply by SW-Niko** [reply]
Danke für die Info, Das Video zeigt das Jk und Victron Smart Shunt ungefähr die gleiche Abweichung, wenn auch mit unterschiedlichen Vorzeichen, haben. Kannst du noch sagen was die kleinsten Ströme sind, was das JK noch messen kann oder zu mindestens anzeigt?

**Comment by schlimmchen** [comment, marked as answer]
Ich hab das Video nicht angeschaut. Meine Erfahrungen sind anders: Meine zwei "old-style BMS" JK-B* messen überhaupt erst ab irgendetwas um die 100mA, wenn ich mich recht erinnere. Jedenfalls enttäuschend spät. Und mit enttäuschender Auflösung. Bei geringen Strömen springt die Messung oft zwischen z.B. -1 A und +1 A hin und her, was mir nicht realistisch erscheint.

Meiner Meinung nach taugt diese Messung am JK-P* BMS nicht. Wie es mit den Inverter-BMS bestellt ist, weiß ich nicht. Ich habs kurz ausprobiert und ich meine mich zu erinnern, dass es immerhin besser ist, aber immer noch nicht (mich) überzeugend.

**Reply by SW-Niko** [reply]
Das habe ich mir schon gedacht. Es ist nicht einfach eine Strommessung zu designen die von 100A bis 0.1A mit einer Toleranz von 5% Messen kann. Das geht aber kostet halt ein paar €.

Dazu kommt noch der Mischstrom den der Inverter zieht. 🤔 
Ich denke die 35€ werde ich investieren und das ganze mal ausprobieren.

Vielen Dank für die Infos. 👍

**Reply by spcqike** [reply]
> Es ist nicht einfach eine Strommessung zu designen die von 100A bis 0.1A mit einer Toleranz von 5% Messen kann.

hm, ich denke das sollte heutzutage keine größe Hürde sein. +-100A mit 0.1A Auflösung sind nicht mal 12bit Auflösung. ein passender Shunt, AD-Wandler (mit internen oder externen PGA) und gut. mit mehrstufigen PGAs bekommt man dann auch im Kleinlastbereich noch sehr gute Ergebnisse und vor allem auch größere Auflösungen, also nur 100mA.

mit einem INA226 und einem passenden Shunt sollte das gut klappen.

Es gibt auch DIY Smartshunt Projekte mit diesen Bauteilen.
https://github.com/kscholty/SmartShuntINA226

Die sollten ggf. sogar hier eingebunden werden können.

**Reply by SW-Niko** [reply]
>12bit Auflösung

5% von 0.1A = 5mA, Von 100A bis 5m ist der Faktor 20000. Mit +- sind es 40000.  16Bit sind 2^16 = 65536.
Man benötigt also einen 16Bit Wandler. Aber du hast Recht, ob 12bit oder 16bit, der Wandler ist nicht das Problem.

>passenden Shunt

Wenn wir bei 5mA noch ein vernünftiges Signal über dem "Rauschteppich" haben wollen, sagen wir 0.01mV dann benötigen wir einen Shunt mit 0.01mV/5mA = 2mOhm. Bei 100A haben wir dann 0.2V und 20W Verlustleistung. Also besser 1mOhm.

Habe ich mich irgendwo verrechnet? 
Ja ... das ist alles technisch machbar. Aber so ein paar Berechnungen zeigen warum die Victron Shunts so große Klötze sind.  🤔

**Reply by spcqike** [reply]
> Habe ich mich irgendwo verrechnet?

Nein. Aber im vorherigen Post vielleicht einfach missverständlich ausgedrückt. Toleranz ist nicht Auflösung. Zumindest nicht für mich (: 

+-100A mit 0.005A Auflösung ist natürlich was anderes, als +-100A und 0.1A Auflösung. Da hast du recht.

**Comment by SW-Niko** [comment]
Ich habe mir nun doch einen Victron Smart Shunt bestellt. 
Im Datenblatt findet man Angaben zur Strommessung und zur Genauigkeit. Was noch fehlt ist eine Angabe zur Zeitkonstante der Strommessung. Mal schauen ... wie der Shunt mit dem DC-AC Mischstrom, den der Inverter zieht, zurechtkommt. 🤔

**Reply by gitisgreat2023** [reply]
Welches Modell hast du genommen? Du meinst mit der Zeitkonstante was die "bandwith" ist, die er abdeckt, richtig? 
Ich bin gespannt wie gut der shunt ist. Schreib hier doch mal rein was deine Erfahrungen sind wenn er mal einige Monate in Betrieb war.

Ich baue demnächst auch einen DIY Akku mit JK BMS. Meine Strategie wird: 1) tätsächliche Ah eintragen und 2) A und V bei mittels Labornetzteil kalibrieren. Wegen Tibber wird er dann auch im Winter regelmässig auf 100% aufgeladen, also auf Dauer stört höchstens einen short-term SoC accuracy error.

**Reply by SW-Niko** [reply]
>Welches Modell hast du genommen? 

Die 500A Variante, War im Angebot

>Du meinst mit der Zeitkonstante was die "bandwith" ist, die er abdeckt, richtig?

Ich meine die Integrationszeitkonstante für den AC Anteil im Mischstrom. Ich will ja nicht das der Stromwert, trotz konstanter Ausgangsleistung vom Inverter, hin und her schwankt.

**Comment by stamartin** [comment]
Ich habe das JK-PB2A16S-15P . Also Max. Strom von 150A.  Die Anzeige schwankt bei mir mit 0,37A.
Ob das die absolute Genauigkeit kann ich aber nicht sagen. In der App ist das der gleiche Wert.

**Reply by SW-Niko** [reply]
Danke für die Info. Schon ein etwas grober Wert, aber für die meisten Anwendungen ausreichend. Ich habe mir jetzt einen SmartShunt bestellt, der hat eine Auflösung von 0,01A.

**Comment by SW-Niko** [comment]
Zur Info.
Smart Shunt ist eingebaut. Hat alles auf Anhieb funktioniert.
Der Shunt und 2 MPPT sind jeweils über einen ADuM1201 an den ESP32-S3 angebunden.
Vergleichsmessungen mit hochwertigen Messgeräten habe ich nicht mehr gemacht. Ich denke da kann ich mich auf die Datenblattangaben von Victron verlassen. 
Die Messzeitkonstante im Shunt ist hoch genug, so das bei dem Mischstrom den der Inverter zieht, die Messwerte sehr stabil bleiben.

**Reply by gitisgreat2023** [reply]
Kannst du bitte Mal berichten wie deine Erfahrungen mit dem smartshunt bisher sind?

**Reply by SW-Niko** [reply]
Hallo @gitisgreat2023,
der Smart Shunt funktioniert in meinem System einwandfrei.
Er hat eine Messauflösung von 1mV / 1mA. Die Daten kommen jede Sekunde.
Er hat sich wunderbar auf meine Batterie einstellen lassen.
Wenn die Batterie 1 Monat lang nicht vollgeladen wird dann weicht der SoC um ca. 3-5%  vom wirklichen Wert ab.
Das kann man noch besser einstellen aber für meine Zwecke reicht das vollkommen aus.
Benötigst du sonst noch welche Infos?

**Reply by swingstate** [reply]
Dito, der Victron Shunt funktioniert bestens mit OpenDTU-OnBattery. Hatte bisher folgende Configs damit laufen: 

1x 25,6V für SoC und Spannung
2x 25,6V in Serie für SoC und Spannung inkl. Mittelspannungsmessung.

**Reply by spcqike** [reply]
Hat schon mal jemand einen Smart shunt selber gebaut oder wurden bisher immer originale gekaufte genutzt?

**Reply by gitisgreat2023** [reply]
> Hallo @gitisgreat2023, der Smart Shunt funktioniert in meinem System einwandfrei. Er hat eine Messauflösung von 1mV / 1mA. Die Daten kommen jede Sekunde. Er hat sich wunderbar auf meine Batterie einstellen lassen. Wenn die Batterie 1 Monat lang nicht vollgeladen wird dann weicht der SoC um ca. 3-5% vom wirklichen Wert ab. Das kann man noch besser einstellen aber für meine Zwecke reicht das vollkommen aus. Benötigst du sonst noch welche Infos?

Ja, die 3-5% interessieren mir. 1 Monat quasi hin un her cycled, zB zwischen 20% und 50%, oder ähnliches?

**Reply by SW-Niko** [reply]
> 1 Monat quasi hin un her cycled, zB zwischen 20% und 50%, oder ähnliches?

Nö, so weit ist bei mir der SoC bis jetzt nicht weggelaufen. 
Aber, ich habe nur einen begrenzten Arbeitsbereich. Maximal 20A und höchstens 1 Monat ohne Vollladung.
Bei längeren Zeiträumen ohne Vollladung wird der sicherlich noch weiter weglaufen.

>Hat schon mal jemand einen Smart shunt selber gebaut oder wurden bisher immer originale gekaufte genutzt?

Also ich nicht.
